src/Sdk/Validation/Validator.php

```diff
@@ -0,0 +1,175 @@
+<?php
+declare(strict_types=1);
+
+namespace LoyaltyCorp\SdkBlueprint\Sdk\Validation;
+
+use LoyaltyCorp\SdkBlueprint\Sdk\DataTransferObject;
+use LoyaltyCorp\SdkBlueprint\Sdk\Exceptions\InvalidRulesException;
+use LoyaltyCorp\SdkBlueprint\Sdk\Exceptions\UndefinedValidationRuleException;
+
+class Validator
+{
+    /**
+     * Repository validation rules
+     *
+     * @var mixed[]
+     */
+    public $rules = [];
+
+    /**
+     * Validate a repository
+     *
+     * @param mixed[] $data The data to validate
+     *
+     * @return string[]
+     */
+    private function validateAttributeValues(array $data): array
+    {
+        // Reset errors
+        $errors = [];
+
+        foreach ($this->rules as $attribute => $rules) {
+            if (\array_key_exists($attribute, $data) === false) {
+                continue;
+            }
```

Comment:
We need to confirm but I think that `isset()` cost is less than `instanceof` so for performance purposes it should be the first statement.